airflow/utils/types.py

```diff
@@ -24,6 +24,9 @@ class DagRunType(str, enum.Enum):
     SCHEDULED = "scheduled"
     MANUAL = "manual"
 
+    def __str__(self) -> str:  # pylint: disable=invalid-str-returned
```

Comment:
Do we need this if it also inherits from `str`?

Reply:
I think this is what the PR actually fixes or intends to do since we were inheriting from str since https://github.com/apache/airflow/pull/11621

Reply:
On current master

```
In [19]: from  airflow.utils.types import DagRunType                                                                                  

In [20]: f'{DagRunType.SCHEDULED}__a'                                                                                                 
Out[20]: 'scheduled__a'
```

Since it inherits from str I don't think we need this method.

(This PR has gone through a number of iterations, firstly removing the `str` subclassing, then adding it back etc, so I'm not 100% sure on where it's at.)

Reply:
we need this, inheriting from str doesn't override `__str__`, but only `__format__`. If you look into the mysql client code i linked in the discussion, `__str__` is what it calls to serialize the binding params:

For example:

```
>>> from  airflow.utils.types import DagRunType
>>> str(DagRunType.SCHEDULED)
'DagRunType.SCHEDULED'
>>> f'{DagRunType.SCHEDULED}'
'scheduled'
>>> format(DagRunType.SCHEDULED)
'scheduled'
```

You can also reproduce the problem with the unit test provided in this PR by removing `__str__` override.

Reply:
Ahhhhh. Gotcha. Nice find.